cloudinit/net/__init__.py

```diff
@@ -133,6 +147,52 @@ def master_is_openvswitch(devname):
     return os.path.exists(ovs_path)
 
 
+@functools.lru_cache()
```

Comment:
Nit: `maxsize=None` (and on the next function)

Reply:
Initially I was going to say:

> Why? Per [the docs](https://docs.python.org/3/library/functools.html#functools.lru_cache):
>
> > If maxsize is set to None, the LRU feature is disabled and the cache can grow without bound.
>
> These functions don't take parameters, so I think the size of the cache will be at most 1.

but I've answered my own question: the `functools` code [special-cases `None`](https://github.com/python/cpython/blob/67148254146948041a77d8a2989f41b88cdb2f99/Lib/functools.py#L549-L562) to make it much simpler; TIL, thanks!